packages/cli/src/commands/git/connect.ts

```diff
@@ -57,6 +60,23 @@ export default async function connect(
   // get project from .git
   const gitConfigPath = join(path, '.git/config');
   const gitConfig = await parseGitConfig(gitConfigPath, output);
+
+  if (repoArg) {
+    if (gitConfig) {
+      return await connectArgWithLocalGit(
+        client,
+        output,
+        org,
+        project,
+        confirm,
+        gitConfig,
+        repoArg
+      );
+    } else {
+      return await connectArg(client, output, confirm, org, project, repoArg);
```

Comment:
Unless others have opinions, you don't need the `else` and instead keep the code left.